spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpSessionFactoryTests.java

```diff
@@ -215,7 +225,39 @@ void customTimeoutIsApplied() throws IOException {
 			ClientChannel clientChannel = sftpSessionFactory.getSession().getClientInstance().getClientChannel();
 
 			assertThat(AbstractSftpClient.SFTP_CLIENT_CMD_TIMEOUT.getRequired(clientChannel)).hasSeconds(15);
+
+			sftpSessionFactory.destroy();
 		}
 	}
 
+	@Test
+	void clientSessionIsClosedOnSessionClose() throws Exception {
+		try (SshServer server = SshServer.setUpDefaultServer()) {
+			server.setPasswordAuthenticator((arg0, arg1, arg2) -> true);
+			server.setPort(0);
+			server.setKeyPairProvider(new SimpleGeneratorHostKeyProvider(new File("hostkey.ser").toPath()));
+			server.setSubsystemFactories(Collections.singletonList(new SftpSubsystemFactory()));
+			server.start();
+
+			DefaultSftpSessionFactory sftpSessionFactory = new DefaultSftpSessionFactory();
+			sftpSessionFactory.setHost("localhost");
+			sftpSessionFactory.setPort(server.getPort());
+			sftpSessionFactory.setUser("user");
+			sftpSessionFactory.setPassword("pass");
+			sftpSessionFactory.setAllowUnknownKeys(true);
+
+			SftpSession session = sftpSessionFactory.getSession();
+			ClientSession clientSession = session.getClientInstance().getClientSession();
+
+			assertThat(session.isOpen()).isEqualTo(true);
+			assertThat(clientSession.isOpen()).isEqualTo(true);
+
+			session.close();
+
+			assertThat(session.isOpen()).isEqualTo(false);
+			assertThat(clientSession.isClosed()).isEqualTo(true);
```

Comment:
Nit-pick.
Isn't there something like `isTrue()/isFalse()` convenient assertions?

Reply:
Yes, there is will push with that change.

Thanks